src/rust/src/contract.rs

```diff
@@ -33,7 +34,7 @@ impl From<CxxLedgerInfo> for LedgerInfo {
             protocol_version: c.protocol_version,
             sequence_number: c.sequence_number,
             timestamp: c.timestamp,
-            network_passphrase: c.network_passphrase,
+            network_id: c.network_id.try_into().unwrap(),
```

Comment:
Is panicking here safe? Could we avoid this by storing a `[u8; 32]` in `CxxLedgerInfo`?

Reply:
Yeah, it should be safe (unless there is some bug in core which causes us passing an incorrect value, at which point panicing is fine).